engine/src/main/java/org/terasology/world/block/family/RegisterBlockFamily.java

```diff
@@ -21,6 +21,10 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * This annotation is used to mark a {@link BlockFamily} to be registered into the game.
```

Comment:
look at RegisterSystem annotation. The behavior is fairly similar. You can provide some examples of the usage.